src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2606,6 +2562,15 @@ Node TheoryStringsRewriter::rewritePrefixSuffix(Node n)
   {
     val = NodeManager::currentNM()->mkNode(kind::MINUS, lent, lens);
   }
+
+  // Check if we can turn the prefix/suffix into equalities by showing that the
+  // prefix/suffix is at least as long as the string
+  Node eqs = inferEqsFromContains(n[0], n[1]);
```

Comment:
We should do something similar here for contains (as I mention in my other comment).  The idea is that contains/prefix whose length entailment holds rewrites to an equality, then subsequently equalities may rewrite to conjunction of equalities in a separate step.